frame/name-service/src/commit_reveal.rs

```diff
@@ -0,0 +1,219 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Handles the commit and reveal part of the registration process.
+
+use crate::{types::*, *};
+use frame_support::traits::{
+	BalanceStatus, Currency, ExistenceRequirement, OnUnbalanced, ReservableCurrency,
+	WithdrawReasons,
+};
+use frame_system::pallet_prelude::BlockNumberFor;
+use sp_runtime::traits::{Saturating, Zero};
+use sp_std::prelude::*;
+
+impl<T: Config> Pallet<T> {
+	/// Get the commitment hash from the raw name and secret.
+	pub fn commitment_hash(name: &[u8], secret: u64) -> CommitmentHash {
+		sp_io::hashing::blake2_256(&(name, secret).encode())
+	}
+
+	/// Get the name hash from raw bytes.
+	pub fn name_hash(name: &[u8]) -> NameHash {
+		sp_io::hashing::blake2_256(name)
+	}
+
+	/// Returns a commitment by hash if it exists.
+	pub fn get_commitment(
+		commitment_hash: CommitmentHash,
+	) -> Result<CommitmentOf<T>, DispatchError> {
+		Commitments::<T>::get(commitment_hash).ok_or(Error::<T>::CommitmentNotFound.into())
+	}
+
+	/// Checks whether a commitment has passed the minimum commitment period.
+	pub fn is_commitment_valid(
+		commitment: &CommitmentOf<T>,
+		block_number: &BlockNumberFor<T>,
+	) -> bool {
+		&commitment.when.saturating_add(T::MinCommitmentAge::get()) <= block_number
+	}
+
+	/// Checks whether a commitment has passed the commitment expiry time.
+	pub fn is_commitment_expired(
+		commitment: &CommitmentOf<T>,
+		block_number: &BlockNumberFor<T>,
+	) -> bool {
+		&commitment.when.saturating_add(T::MaxCommitmentAge::get()) <= block_number
+	}
+
+	/// Handles a new commitment for a name hash.
+	///
+	/// When `MinCommitmentAge` blocks have passed, any user can submit `reveal` with the
+	/// `name` and `secret` parameters, and the registration will be completed.
+	///
+	/// See `fn reveal`.
+	pub fn do_commit(
+		depositor: T::AccountId,
+		owner: T::AccountId,
+		commitment_hash: CommitmentHash,
+	) -> DispatchResult {
+		ensure!(!Commitments::<T>::contains_key(commitment_hash), Error::<T>::CommitmentExists);
+
+		let maybe_deposit = CommitmentDeposit::<T>::get();
```

Comment:
A cleaner way might be:

```suggestion
		let deposit = CommitmentDeposit::<T>::get().ok_or(Error::<T>::CommitmentsDisabled)?;
```

To get rid of the `ensure!` and `expect`.

Reply:
Ah yes, good spot!